database/mongodb/mongodb.go

```diff
@@ -0,0 +1,160 @@
+package mongodb
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database"
+	"github.com/mongodb/mongo-go-driver/bson"
+	"github.com/mongodb/mongo-go-driver/mongo"
+	"github.com/mongodb/mongo-go-driver/x/bsonx"
+	"github.com/mongodb/mongo-go-driver/x/network/connstring"
+)
+
+func init() {
+	database.Register("mongodb", &Mongo{})
+}
+
+var DefaultMigrationsTable = "schema_migrations"
+
+var (
+	ErrNoDatabaseName = fmt.Errorf("no database name")
+	ErrNilConfig      = fmt.Errorf("no config")
+)
+
+type Mongo struct {
+	client *mongo.Client
+	db     *mongo.Database
+
+	config *Config
+}
+
+type Config struct {
+	DatabaseName         string
+	MigrationsCollection string
+}
+
+type versionInfo struct {
+	Version int  `bson:"version"`
+	Dirty   bool `bson:"dirty"`
+}
+
+func WithInstance(instance *mongo.Client, config *Config) (database.Driver, error) {
+	if config == nil {
+		return nil, ErrNilConfig
+	}
+	if len(config.DatabaseName) == 0 {
+		return nil, ErrNoDatabaseName
+	}
+	if len(config.MigrationsCollection) == 0 {
+		config.MigrationsCollection = DefaultMigrationsTable
+	}
+	mc := &Mongo{
+		client: instance,
+		db:     instance.Database(config.DatabaseName),
+		config: config,
+	}
+	return mc, nil
+}
+
+func (m *Mongo) Open(dsn string) (database.Driver, error) {
+	uri, err := connstring.Parse(dsn)
```

Comment:
I like using the mongodb's connection string parsing code, but I'm worried that it's classified as experimental/unstable.

But it's probably cleaner to use use mongodb's experimental code than it is to write our own custom db name extraction code, so it's fine to leave this as is.

Reply:
I think that we should let this parse to [mongo-go-driver](https://github.com/mongodb/mongo-go-driver), it use same function for parse connection to mongodb (atleast for now). I'll add this info to readme in next itteration.